physical/aerospike/aerospike.go

```diff
@@ -0,0 +1,155 @@
+package aerospike
+
+import (
+	"context"
+	"fmt"
+	"hash/fnv"
+	"strconv"
+	"strings"
+
+	aero "github.com/aerospike/aerospike-client-go"
+	log "github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/vault/sdk/physical"
+)
+
+const (
+	keyBin   = "keyBin"
+	valueBin = "valueBin"
+)
+
+// AerospikeBackend is a physical backend that stores data in Aerospike.
+type AerospikeBackend struct {
+	client    *aero.Client
+	namespace string
+	set       string
+	logger    log.Logger
+}
+
+// Verify AerospikeBackend satisfies the correct interface.
+var _ physical.Backend = (*AerospikeBackend)(nil)
+
+// NewAerospikeBackend constructs an AerospikeBackend backend.
+func NewAerospikeBackend(conf map[string]string, logger log.Logger) (physical.Backend, error) {
+	namespace := conf["namespace"]
+	set := conf["set"]
+
+	policy := aero.NewClientPolicy()
+	policy.User = conf["username"]
+	policy.Password = conf["password"]
+
+	port, err := strconv.Atoi(conf["port"])
```

Comment:
If `port` is not provided, the error returned does not make it obvious that this is the case. Consider checking the value beforehand separately.

```
strconv.Atoi: parsing "": invalid syntax
```